### PR TITLE
clean up tracing charts

### PR DESCRIPTION
**Pull request by rshriram**
The tracing charts for jaeger and zipkin have an assortment of helm settings
that are getting out of hand and not being maintained properly. In the end,
all of these settings are translated into zipkin/jaeger specific environment variables.

This PR removes all these custom settings and converts them into a simplified env
field in zipkin/jaeger. By doing so, we expose all options in Zipkin/Jager server to end users.
They can tweak the system to their heart's content without having to keep tweaking the tracing charts.

Signed-off-by: Shriram Rajagopalan <rshriram@tetrate.io>

### CODE REVIEW

File: install/kubernetes/helm/istio/charts/tracing/templates/deployment-jaeger.yaml [hunk 1]
```diff
@@ -59,22 +59,12 @@ spec:
...
-          {{- if eq .Values.jaeger.spanStorageType "badger" }}
-          - name: BADGER_EPHEMERAL
-            value: "false"
-          - name: SPAN_STORAGE_TYPE
-            value: "badger"
-          - name: BADGER_DIRECTORY_VALUE
-            value: "/badger/data"
-          - name: BADGER_DIRECTORY_KEY
-            value: "/badger/key"
+          {{- if .Values.env }}
```
Comment (objectiser) [inline install/kubernetes/helm/istio/charts/tracing/templates/deployment-jaeger.yaml]: Shouldn't this be `.Values.jaeger.env`?

File: install/kubernetes/helm/istio/charts/tracing/templates/deployment-jaeger.yaml [hunk 2]
```diff
@@ -83,11 +73,9 @@ spec:
             httpGet:
               path: /
               port: 16686
-{{- if eq .Values.jaeger.spanStorageType "badger" }}
```
Comment (objectiser) [inline install/kubernetes/helm/istio/charts/tracing/templates/deployment-jaeger.yaml]: Isn't ideal for the volume mounts and volumes - is there a way to check the value of a map entry, i.e. `eq .Values.jaeger.env['SPAN_STORAGE_TYPE'] "badger"`?

File: install/kubernetes/helm/istio/charts/tracing/templates/deployment-zipkin.yaml [hunk 3]
```diff
@@ -61,14 +61,12 @@ spec:
...
-            - name: QUERY_PORT
-              value: "{{ .Values.zipkin.queryPort }}"
-            - name: JAVA_OPTS
-              value: "-XX:ConcGCThreads={{ .Values.zipkin.node.cpus }} -XX:ParallelGCThreads={{ .Values.zipkin.node.cpus }} -Djava.util.concurrent.ForkJoinPool.common.parallelism={{ .Values.zipkin.node.cpus }} -Xms{{ .Values.zipkin.javaOptsHeap }}M -Xmx{{ .Values.zipkin.javaOptsHeap }}M -XX:+UseG1GC -server"
-            - name: STORAGE_METHOD
-              value: "mem"
-            - name: ZIPKIN_STORAGE_MEM_MAXSPANS
-              value: "{{ .Values.zipkin.maxSpans }}"
+            {{- if .Values.env }}
+            {{- range $key, $val := .Values.env }}
```
Comment (morvencao) [inline install/kubernetes/helm/istio/charts/tracing/templates/deployment-zipkin.yaml]: should be `.Values.zipkin.env`

File: install/kubernetes/helm/istio/charts/tracing/templates/deployment-jaeger.yaml (hunk 2 above)
Comment (rshriram) [reply install/kubernetes/helm/istio/charts/tracing/templates/deployment-jaeger.yaml]: I am not sure. But also, keep in mind that this is essentially mounting an empty directory at the cost of a big simplification in the UX and flexibility. No persistent volumes,etc.

File: install/kubernetes/helm/istio/charts/tracing/values.yaml [hunk 4]
```diff
@@ -32,35 +32,37 @@ jaeger:
...
   persist: false
-  storageClassName: ""
-  accessMode: ReadWriteMany
+  env:
+    SPAN_STORAGE_TYPE: "badger"
+    BADGER_EPHEMERAL: "false"
+    BADGER_DIRECTORY_VALUE: "/badger/data"
+    BADGER_DIRECTORY_KEY: "/badger/key"
+    COLLECTOR_ZIPKIN_HTTP_PORT: "9411"
+    MEMORY_MAX_TRACES: 50000
```
Comment (objectiser) [inline install/kubernetes/helm/istio/charts/tracing/values.yaml]: This `MEMORY_MAX_TRACES` option could probably be left out, so options only relate to the storage type that has been selected. If a user wants memory storage, then they would need to define other options (and probably would remove the badger related ones).

File: install/kubernetes/helm/istio/charts/tracing/templates/deployment-jaeger.yaml (hunk 2 above)
Comment (objectiser) [reply install/kubernetes/helm/istio/charts/tracing/templates/deployment-jaeger.yaml]: Fair point.

**Comment by objectiser** [comment]
@rshriram Will these changes be mirrored to `istio-installer`? Not clear when work on these charts is going to stop?

**Comment by rshriram** [comment]
oh they should be mirrored there as well. I haven't kept up with the installer. That said, I did see changes going into these charts a day or two ago. So I thought I would clean them up.

**Comment by sdake** [comment]
@rshriram please  try to merge into istio/installer first.  We are really working hard to make that the upstream of all installer work.  If this work merges there, we can merge the same here.

We require an upgrade path from the old (this) to the new (installer repo) and changing these things could have upgrade impact (that can be evaluated more effectively by the environments team which are really focused on reviewing the installer repo.

Treat this directory as a downstream of the upstream istio/installer repo.

Hope that helps.

Cheers
-steve

**Comment by costinm** [comment]
Also: I would also tie any backward incompatible changes to tracing to moving it out of istio-system ( which can only be done with new installer and operator ). 

And what we have been discussion is to clean up values.yaml and installer by moving away anything that can be kustomized ( i.e. can be implemented by applying a patch or by composition). Many of the volumes mounts and env variables fit in this category. At the same time move any defaults or mainstream configs to a proper istio API (telemetry or networking) instead of installer.

**Comment by costinm** [comment]
I should also ask again anyone in OWNERs who keeps approving changes to old installer to stop letting in changes that are not first done in old installer ( or have a PR open at least ), and as was always the case to block and backward-compat breaking change. The 1.1 nightmare was enough.

My focus has been on the new installer - with the assumption that old installer will stay stable.

There is work in operator to have a proper .proto for values.yaml and proposal to move it to istio/api, to increase the review - obviously this is likely out of sync given the changes in master.

**Pull request closed without merging**